### ISSUE TITLE
golangci-lint in super-linter produces false positives

### ISSUE THREAD

**Issue by nantiferov** [super-linter/super-linter#3596]
## Describe the bug

Basically the same as closed as completed previous [issue #1599](https://github.com/github/super-linter/issues/1599)

It seems that this haven't fixed previous issue with https://github.com/github/super-linter/pull/3043

I just tried latest v4.9.7 both standard and slim images and they show errors for my code separated to multiple files for one package.

## Expected behavior

golang-ci scan do not produce false positive errors.

## Steps to Reproduce

1. Create two files in one folder: main.go and foo.go

```main.go
package main

var test int

func main() {
	foo()
}
```

```foo.go
package main

import "fmt"

func foo() {
	fmt.Println(test)
}
```

2. Run golangci-lint for each file separately without --fast option and get errors as expected (logs below)
3. Then add --fast option and all will be fine (logs below)
4. Copy [config used in super-linter](https://github.com/github/super-linter/blob/main/TEMPLATES/.golangci.yml) and add it to command line, the same way as it done in super-linter https://github.com/github/super-linter/blob/main/lib/linter.sh#L897 and it will produce error, even with --fast option (logs below)

I don't know how to fix this in correct way. Definitely all works fine if `.golangci.yml` is not passed.

## Logs

```log
# without fast option
$ golangci-lint run main.go 
main.go:6:2: undeclared name: `foo` (typecheck)
        foo()
        ^
$ golangci-lint run foo.go
foo.go:6:14: undeclared name: `test` (typecheck)
        fmt.Println(test)
                    ^

# with --fast
$ golangci-lint run --fast main.go
$ golangci-lint run --fast foo.go 

# with .golangci.yml
$ golangci-lint run --fast main.go  
main.go:6:2: undeclared name: `foo` (typecheck)
        foo()
        ^
$ golangci-lint run --fast foo.go  
foo.go:6:14: undeclared name: `test` (typecheck)
        fmt.Println(test)
                    ^
```

## Additional context

`golangci-lint` version 1.50.1 (also tested v1.49.0 with the same results)

I think it might be related to this issue https://github.com/golangci/golangci-lint/issues/1909

**Comment by erolkskn** [comment]
+1

**Comment by ferrarimarco** [comment]
Perhaps it would be enough to remove the `--fast` option from our invocation?

From the [golangci-lint docs](https://golangci-lint.run/usage/configuration/#command-line-options):

```
Run only fast linters from enabled linters set (first run won't be fast)
```

This way, super-linter users have full control over the linters that run via the `golangci-lint` configuration file.

**Comment by erolkskn** [comment]
@ferrarimarco Unfortunately it wouldn't :disappointed: That's because of `--fast` flag also specifies that only linters that are able to handle *single file* should be ran. (See: https://github.com/golangci/golangci-lint/issues/1574#issuecomment-804504899). Since **super-linter** tries to lint files one by one, we kinda need that flag. 

On the other hand, currently `--fast` option can't override options specified in the config file. (See: https://github.com/golangci/golangci-lint/issues/1909). That's probably the root cause of why go linting doesn't properly work with in super-linter. 

About the error on this issue, `typecheck` linter can't handle single file and it shouldn't be in use while linting single file. `--fast` option ought to be filter that when passed but clearly it can't. That's why linting fails.

**Comment by DiogoMCampos** [comment]
I am facing the same issue, and although I appreciate the direct issue here is that the `--fast` option doesn't quite work as expected and does not override, but is in fact overridden by, other options, I think running the linter on a file-per-file basis simply isn't adequate for Go projects beyond a few dozens of lines.
